### PR TITLE
yay datatables

### PR DESCRIPTION
**Pull request by Galithil**
allows javascript querying for fields in the tables, rather than all table wide search.
Yes, the same could be accomplished if we had a SQL db and some proper queries.

### CODE REVIEW

File: run_dir/design/worksets.html [hunk 1]
```diff
@@ -36,6 +30,6 @@ <h1 id="{{ worksets }}" class="worksets_page_heading"><span id="page_title">All
 
 </div> <!-- #page_content -->
 
-<script src="/static/js/list.min.js"></script>
+<script src="/static/js/jquery.dataTables.min.js?v={{ gs_globals['git_commit'] }}"></script>
```
Comment (ewels) [inline run_dir/design/worksets.html]: I don't think we need the `?v={{ gs_globals['git_commit'] }}` for libraries that we're including, do we? This code shouldn't be changing much.

Mind you, I guess it doesn't really hurt either.

**Comment by ewels** [comment]
Super nice :+1: Only things I can think of are those minor presentation things I mentioned yesterday - pulling the main search box up into the header mostly. That doesn't really matter though. Also might be nice to try to generalise and consolidate the code into `base.js` so that we don't have three copies to maintain? Neither are show stoppers though, so good to merge if you don't have the time / disagree.

**Comment by Galithil** [comment]
I don't really know how to send the general search box in the header
because as I've said, it's generated by the lib. changing its place
afterwards might be an odd thing to do and might break css/js triggers or
things like that. How would you have a go at it ?

I agree about generalizing the code, but that will have to wait a little
bit, unfortunately.

**Comment by ewels** [comment]
Just one extra line of javascript, similar to where you'd adding the CSS classes I think. Something like this:
```javascript
$("#fc_table_filter").appendTo("h1");
```
Only caveat is that it _may_ break it if it's not longer inside some parent object, as you say. But easiest to just try it and see if it still works.

**Comment by ewels** [comment]
Excellent work :+1:

**Pull request merged**